components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/SAMLAssertionClaimsCallback.java

```diff
@@ -192,7 +203,9 @@ public void handleCustomClaims(JWTClaimsSet jwtClaimsSet, OAuthAuthzReqMessageCo
         } else {
             claims = getClaimsMap(userAttributes);
         }
-        return claims;
+        String tenantDomain = requestMsgCtx.getAuthorizedUser().getTenantDomain();
```

Comment:
tenant domain needs to be that of the SP to handle SaaS and federated scenarios.